pkg/notifications/stream.go

```diff
@@ -0,0 +1,40 @@
+package notifications
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/stackrox/rox/generated/storage"
+)
+
+const bufferSize = 100
+
+// Stream is an interface for the notifications stream.
+type Stream interface {
+	Consume() <-chan *storage.Notification
+	Produce(event *storage.Notification) error
+}
+
+func newStream() Stream {
+	return &streamImpl{notificationChan: make(chan *storage.Notification, bufferSize)}
+}
+
+type streamImpl struct {
+	notificationChan chan *storage.Notification
+}
+
+// Consume returns the channel to retrieve notifications.
+func (s *streamImpl) Consume() <-chan *storage.Notification {
+	return s.notificationChan
+}
+
+// Produce adds a notification to the stream.
+//
+// Should be retried if an error is returned.
```

Comment:
Instead of only documenting this, we can also ensure that the error is retryable via [retry.MakeRetryable](https://github.com/stackrox/stackrox/blob/master/pkg/retry/retryable.go#L5).

Then clients may benefit from `retry.WithRetry(s.Produce(notifications)` ([see here](https://github.com/stackrox/stackrox/blob/master/pkg/retry/retry.go#L6)). That'd make things easier.